jasmine-standalone-2.5.2/spec/invertedIndex.spec.js

```diff
@@ -0,0 +1,211 @@
+    });
+
+    it('should return true if index is created', () => {
+      expect(invertedIndex.createIndex('books.json', book)).toEqual(true);
+    });
+  });
+
+  describe('getIndices', () => {
+    it('should verify that index has been created', () => {
+      expect(Object.keys(invertedIndex.getIndices('books.json')).length)
+      .toBeGreaterThan(0);
+    });
+
+    it('should check that index maps the string to the correct objects in json'
+     + ' array', () => {
+      const createdIndex = {
+        a: [0],
+        alice: [0],
+        and: [0],
+        enters: [0],
+        falls: [0],
+        full: [0],
+        hole: [0],
+        imagination: [0],
+        in: [0],
+        into: [0],
+        of: [0],
+        rabbit: [0],
+        wonderland: [0],
+        world: [0]
+      };
+      let result = {};
+      result = invertedIndex.getIndices('anotherBook.json');
+      expect(Object.keys(result)).toEqual(Object.keys(createdIndex));
+      expect(values(result)).toEqual(values(createdIndex));
+    });
+  });
+
+  describe('searchIndices', () => {
+    it('should return true if search term is a string', () => {
+      const words = 'I love Barbie and Alice';
+      expect(Object.keys(invertedIndex.searchIndices(words,
+        'book.json'))).toBeTruthy();
+    });
+
+    it('should return true if search term is a string', () => {
+      const newWords = ['I love Barbie and Alice'];
+      expect(Object.keys(invertedIndex.searchIndices(newWords,
+        'anotherBook.json'))).toBeTruthy();
+    });
+
+    it('should return true if search term is a number', () => {
+      const number = 1234;
+      expect(Object.keys(invertedIndex.searchIndices(number,
+        'anotherBook.json'))).toBeTruthy();
+    });
+
+    it('should search through single files that are indexed', () => {
+      const expectedResult = {
+        0:
+        {
+          barbie: [1],
+          and: [0, 1],
+          cindarella: [1],
+          dearie: [0]
+        }
+      };
+      let search = {};
+      search = invertedIndex.searchIndices('anotherBook.json',
+        'barbie, mercy and cindarella dearie');
+      expect(Object.keys(search)).toEqual(Object.keys(expectedResult));
+      expect(values(expectedResult)).toEqual(values(expectedResult));
+    });
+
+    it('should search through all files', () => {
+      const allFiles =
+        {
+          0:
+          {
+            alice: [0],
+            an: [1],
+            barbie: [2],
+            cartoons: [2],
+            of: [0, 1],
+            unusual: [1],
+            wizard: [1]
+          },
+          1:
+          {
+            barbie: [1]
+          }
+        };
+
+      let search = {};
+      search = invertedIndex.searchIndices('all',
+        'Barbie loves cartoons but she\'s scared of an unusual wizard,' +
+          'alice fall\'s');
+      expect(Object.keys(search)).toEqual(Object.keys(allFiles));
+      expect(values(allFiles)).toEqual(values(allFiles));
+    });
+  });
```

Comment:
What would happen in a situation where a user want to search for 2 files out of 4 uploaded files?